build/teamcity/util/roachtest_util.sh

```diff
@@ -62,8 +65,30 @@ function upload_stats {
   fi
 }
 
-# Upload any stats.json we can find, no matter what happens.
-trap upload_stats EXIT
+set -x
+
+# Uploads roachprod and roachtest binaries to GCS.
+function upload_binaries {
```

Comment:
Since we only run the roachtest nightlies on Linux machines, wouldn't this mean we never publish Mac binaries?

Reply:
Indeed. I don't think it's a deal breaker for this PR, but we may need to find/patch a job that already builds for mac, maybe "local roachtest"?

Reply:
Updated the commit msg. I'll do a quick poll to see how many would prefer that mac and arm64 is also available.